pkg/kubectl/cmd/delete.go

```diff
@@ -306,8 +306,8 @@ func DeleteResult(r *resource.Result, out io.Writer, ignoreNotFound bool, graceP
 }
 
 func cascadingDeleteResource(info *resource.Info, out io.Writer, shortOutput bool) error {
-	falseVar := false
-	deleteOptions := &metav1.DeleteOptions{OrphanDependents: &falseVar}
+	policy := metav1.DeletePropagationForeground
+	deleteOptions := &metav1.DeleteOptions{PropagationPolicy: &policy}
```

Comment:
@kubernetes/sig-cli-api-reviews this is a behavior change, for a resource that doesn't have a reaper, prior to this change, it's deleted immediately by `kubectl delete --cascading=true`, now it's deleted after all its dependents are deleted. The new behavior is more similar to reaper's behavior (e.g., replicaset is deleted after it's scaled down).

Reply:
@liggitt @pwittrock ^^^

Reply:
+1 from me, esp. that this is the current direction we're trying to encourage.